app/src/main/java/com/example/weatherapp/ui/utils/ViewModelExtensions.kt

```diff
@@ -6,10 +6,10 @@ import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.launch
 
-fun ViewModel.onNetworkAvailability(
+suspend fun ViewModel.onNetworkAvailability(
     isAvlb: Boolean,
     onAvailable: () -> Unit,
-    onUnavailable: () -> Unit,
+    onUnavailable: suspend () -> Unit,
```

Comment:
Don't use suspend function here. Instead use normal function and then scope the work you are doing in the place that is invoking it.